app/services/flexible_translation_service.py

```
@@ -0,0 +1,347 @@
+        
+        if provider not in self._providers:
+            raise TranslationError(f"Unknown provider: {provider}. Available: {list(self._providers.keys())}")
+        
+        translation_provider = self._providers[provider]
+        
+        # Check language support
+        if not translation_provider.supports_language_pair(source_lang, target_lang):
+            raise TranslationError(f"Provider {provider} does not support language pair {source_lang}->{target_lang}")
+        
+        try:
+            logger.info(f"Batch translating {len(texts)} texts with {provider} (model: {model or 'default'})")
```

Pylint (3.3.7)

[convention] Line too long (110/100)

(C0301)